### PR TITLE
added estimated time to read

### PR DESCRIPTION
**Pull request by tilast**

### CODE REVIEW

File: app/views/issue_mailer/notify_subscribers.slim [hunk 1]
```diff
@@ -12,6 +12,8 @@ body topmargin="0" leftmargin="0" marginheight="0" marginwidth="0" style="-webki
       - @issue.links.each do |link|
         li
           a href="#{link.url}" target="_blank" style="font-size: 18px; color: #4183c4; text-decoration: none;" #{link.title}
+          - if link.time_to_read
+            p Estimated time to read: #{link.time_to_read}
```
Comment (rilian) [inline app/views/issue_mailer/notify_subscribers.slim]: there will be too many 'estimated time to read ' words in the email. can we render this value after the link text in the same line? maybe in brackets

File: app/views/links/index.slim [hunk 2]
```diff
@@ -22,6 +22,9 @@
               br
               a href="#{link.url}" target="_blank" rel="nofollow" #{link.url}
               br
+              - if link.time_to_read
+                br
+                div Estimated time to read: #{link.time_to_read}
```
Comment (rilian) [inline app/views/links/index.slim]: same as above

**Comment by tilast** [comment]
ok, will fix

**Comment by tilast** [comment]
@rilian fixed

File: app/views/issue_mailer/notify_subscribers.slim [hunk 3]
```diff
@@ -11,7 +11,8 @@ body topmargin="0" leftmargin="0" marginheight="0" marginwidth="0" style="-webki
     ul
       - @issue.links.each do |link|
         li
-          a href="#{link.url}" target="_blank" style="font-size: 18px; color: #4183c4; text-decoration: none;" #{link.title}
+          a href="#{link.url}" target="_blank" style="font-size: 18px; color: #4183c4; text-decoration: none;"
+            | #{link.title} #{link.time_to_read ? "(#{link.time_to_read})" : ''}
```
Comment (rilian) [inline app/views/issue_mailer/notify_subscribers.slim]: is it possible to remove time from the anchor itself?

Comment (tilast) [inline app/views/issue_mailer/notify_subscribers.slim]: like

``` slim
  a href="#{link.url}" target="_blank" style="font-size: 18px; color: #4183c4; text-decoration: none;" #{link.title}
  span #{link.time_to_read ? "(#{link.time_to_read})" : ''}
```

?

Comment (rilian) [inline app/views/issue_mailer/notify_subscribers.slim]: yes

**Comment by rilian** [comment]
can you pl add a final screenshot

**Comment by tilast** [comment]
On my local
index html 2016-01-18 15-32-40

newsletter 2016-01-18 15-33-34

**Comment by rilian** [comment]
let us discuss this later. i would like to talk about ux

**Comment by tilast** [comment]
ok, np

**Comment by tilast** [comment]
new tab 2016-02-01 13-50-33

we can write the text for reading by ourselves, so it can be like this

**Comment by tilast** [comment]
@rilian

**Comment by bambycha** [comment]
@tilast Why you didn't use https://github.com/garethrees/readingtime or something like that?

**Comment by rilian** [comment]
please add a space between words

File: db/migrate/20160118115624_add_estimation_time_to_read.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddEstimationTimeToRead < ActiveRecord::Migration
+  def change
+    add_column :links, :time_to_read, :string
```
Comment (rilian) [inline db/migrate/20160118115624_add_estimation_time_to_read.rb:3]: can we make it integer?

Comment (tilast) [inline db/migrate/20160118115624_add_estimation_time_to_read.rb:3]: what if this article can take several hours to read or something like that? my main aim was to leave the ability to write any estimation

**Comment by tilast** [comment]
@bambycha because estimated time is subjective measurement and it will require load article from its origin, but we don't need this

File: db/migrate/20160118115624_add_estimation_time_to_read.rb (hunk 4 above)
Comment (rilian) [inline db/migrate/20160118115624_add_estimation_time_to_read.rb:3]: in this case, for example if we post a link to a book, we will just skip writing this estimation

File: app/views/links/index.slim [hunk 5]
```diff
@@ -21,6 +21,9 @@
               a href="#{link.url}" target="_blank" rel="nofollow" #{link.title}
               br
               a href="#{link.url}" target="_blank" rel="nofollow" #{link.url}
+              - if link.time_to_read
+                br
```
Comment (rilian) [inline app/views/links/index.slim]: sorry but can we 
1) leave this on the same line with link text
2) keep spaces between words
3) remove link from this estimated time value ?

**Comment by tilast** [comment]
newsletter 2016-02-01 16-26-58

new tab 2016-02-01 16-24-54

**Comment by tilast** [comment]
@rilian done

**Comment by rilian** [comment]
now - nice :+1:

**Pull request merged**